File: CommonCoreLegacy/src/main/java/com/skedgo/android/common/util/CityJsonDeserializer.java
package com.skedgo.android.common.util;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParseException;
import com.skedgo.android.common.model.Region;

import java.lang.reflect.Type;

final class CityJsonDeserializer implements JsonDeserializer<Region.City> {
  @Override
  public Region.City deserialize(JsonElement json,
                                 Type typeOfT,
                                 JsonDeserializationContext context) throws JsonParseException {
    final JsonObject jsonCity = json.getAsJsonObject();
    final String title = jsonCity.getAsJsonPrimitive("title").getAsString();
    final double lng = jsonCity.getAsJsonPrimitive("lng").getAsDouble();
    final double lat = jsonCity.getAsJsonPrimitive("lat").getAsDouble();

    final Region.City city = new Region.City();
    city.setName(title);
    city.setLat(lat);
    city.setLon(lng);
    return city;
  }
}

File: CommonCoreLegacy/src/main/java/com/skedgo/tripkit/common/util/CityJsonDeserializer.kt
package com.skedgo.tripkit.common.util

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.JsonParseException
import com.skedgo.tripkit.common.model.region.Region.City
import java.lang.reflect.Type

internal class CityJsonDeserializer : JsonDeserializer<City> {
    @Throws(JsonParseException::class)
    override fun deserialize(
        json: JsonElement,
        typeOfT: Type,
        context: JsonDeserializationContext
    ): City {
        val jsonCity = json.asJsonObject
        val title = jsonCity.getAsJsonPrimitive("title").asString
        val lng = jsonCity.getAsJsonPrimitive("lng").asDouble
        val lat = jsonCity.getAsJsonPrimitive("lat").asDouble

        val city = City()
        city.name = title
        city.lat = lat
        city.lon = lng
        return city
    }
}